src/kernels/kernelFinder.ts

```diff
@@ -61,7 +61,11 @@ export class KernelFinder implements IKernelFinder {
         const kernels: KernelConnectionMetadata[] = [];
 
         for (const finder of this._finders) {
-            const contributedKernels = finder.listContributedKernels(resource);
+            const contributedKernels = finder.listContributedKernels(resource).map((kernelConnection) => {
+                // Tag our connections with the kernel finder info that located them
+                kernelConnection.kernelFinderInfo = finder;
```

Comment:
I think the `finder` should update the `kernelFinderInfo` as thats the class that creates and returns the `KernelConnection`.
This way no matter who creats these we know it will always be setup.

Reply:
Sounds good. I'll make the change.